python/cudf/cudf/core/join/casting_logic.py

```diff
@@ -183,6 +191,23 @@ def _libcudf_to_output_castrules(lcol, rcol, how):
     if pd.api.types.is_dtype_equal(ltype, rtype):
         return ltype
 
+    if isinstance(ltype, Decimal64Dtype) and isinstance(rtype, Decimal64Dtype):
+        p1, p2 = ltype.precision, rtype.precision
+        s1, s2 = ltype.scale, rtype.scale
+
+        if how == "inner":
+            scale = min(s1, s2)
+            precision = scale + min(p1 - s1, p2 - s2)
```

Comment:
Could we add a small comment explaining why we chose to compute the result scale/precision this way? I'll admit it's not obvious to me :)

Reply:
Ditto below for the `"outer"` case.